Fix-up myriad of items

drewvolz opened the pull request:
* remove duplicate canGoBack prop in home index
* add missing navigation hook dep to contacts list
* export props from report editor
* remove empty navigator file

Recommend [viewing without whitespace](https://github.com/StoDevX/AAO-React-Native/pull/5975/files?w=1)

---

~|~
--|--
ESLint | - 381 problems (0 errors, 381 warnings)<br>+ 380 problems (0 errors, 380 warnings)
TSC | - 339 lines <br> + 304 lines
TSC +- | -3 TS2783<br>-1 TS1208<br>-1 TS2459<br>+2 TS2783

drewvolz committed:
fixup array of items

* remove duplicate canGoBack prop in home index
* add missing navigation hook dep to contacts list
* export props from report editor
* remove empty navigator file

hawkrives commented on source/views/home/index.tsx:
```diff
@@ -91,7 +91,7 @@ export const NavigationKey = 'Home'
 	title: 'All About Olaf',
 	headerBackTitle: 'Home',
-	headerRight: (props) => <OpenSettingsButton canGoBack={true} {...props} />,
+	headerRight: (props) => <OpenSettingsButton {...props} />,
```
Shouldn't we put canGoBack on the right side, to ensure that it's always true?

drewvolz replied:
I think it is already a provided prop on the right side so this would fix a warning about it being added twice. Unless I'm not understanding you correctly and you're saying to replace the value some other way?

hawkrives replied:
Because `props.canGoBack` is a required property, it will always be set, so it will always overwrite our `canGoBack={true}` attribute.

I think I'm proposing that we set it on the right side, after `{…props}` is expanded, so that we ignore the value from `props` and always set it to be `true`.

I don't remember why we put `canGoBack=true` in the first place, but I imagine that somehow we could configure it to be false in `props`… is that really a problem, though?

```tsx
// Current:
	{headerRight: (props) => <OpenSettingsButton canGoBack={true} {...props} />,}
// Proposed:
	{headerRight: (props) => <OpenSettingsButton {...props} />,}
// Alternate Proposal:
	{headerRight: (props) => <OpenSettingsButton {...props} canGoBack={true} />,}
```

hawkrives replied:
Okay. I've convinced myself - let's just do what you have here.

drewvolz replied:
Alright. If we find a need for it we can come back to this!

drewvolz committed:
remove empty home.tsx file

drewvolz committed:
prettier changes to hours overview

drewvolz committed:
opt to use built-in navigation prop instead of navigation hook

drewvolz committed:
update tsc errors/counts and eslint problems

drewvolz committed:
Merge branch 'master' into cleanup

drewvolz committed:
Revert "opt to use built-in navigation prop instead of navigation hook"

This reverts commit 87c57910324050cda2c4ff968da0649ac794aad4.

drewvolz committed:
clean up contact list props and return type

hawkrives committed:
tweak OrgsList

- useAsync -> useFetch (we don't need the added complexity of useAsync)
- fix a typo
- remove custom state variable `isInitialFetch`
